Makefile

```diff
@@ -203,6 +203,7 @@ $(BIN_WINDOWS): generate/zz_filesystem_generated.go
 ##@ Schemas
 ######################
 schema-generate: schema/func_yaml-schema.json ## Generate func.yaml schema
+.PHONY: schema/func_yaml-schema.json
```

Comment:
I am pretty sure that this is not a .PHONY, because it is actually intended to create the file `schema/func_yaml-schema.json.  Did you mean to say `.PHONY schema-generate`?

Reply:
`make build` depends on `$(CODE)` which includes `schema/func_yaml-schema.json`. Since the file exists, `make` does not attempt to regenerate that file. This caused an issue later in the CI pipeline. Since there was a `.PHONY: generate/zz_filesystem_generated.go`, I thought it would be ok to do the same for the schema file. What do you think we should do instead?

Reply:
I think @lkingland is correct here. Modifying `pkg/functions/function.go` does indeed cause the schema to be regenerated via this make target:

https://github.com/knative/func/blob/d5b4a8d8e9bfe8a4ffc66215253f75f1feab7d69/Makefile#L206-L207

The `schema-generate` target should have been a no-op in CI unless the `schema/func_yaml-schema.json` changed during the build. This has not historically been a problem, so IMO this change is not needed. Otherwise all looks good!

Reply:
> Since there was a .PHONY: generate/zz_filesystem_generated.go, I thought it would be ok to do the same for the schema file.

I see; thanks for noticing!  I believe that `.PHONY: generate/zz_filesystem_generated.go` should actually be removed as well.  The way it is being generated is not entirely optimal, so it's probably not the best example.

Here's how I understand this rule to work:
```
schema/func_yaml-schema.json: pkg/functions/function.go 
        go run schema/generator/main.go
```
`pkg/functions/function.go` is the dependency. The rule will only execute if the target file either doesn't exist or is older than the dependency file. In this case, if `function.go` is newer than `func_yaml-schema.json` (has modifications), then the rule will be executed, generating the file again via `go run...`.  This is the intended functionality.

I would like to know the problem you saw in CI.  There actually might be something over there we need to fix or clean up!

Reply:
I know what the issue is now; I added a commit. `schema/func_yaml-schema.json` should also depend on `function_*.go`